src/core.rs

```diff
@@ -113,7 +113,8 @@ impl EncodedPrefixVarInt {
     }
 
     pub fn as_slice(&self) -> &[u8] {
-        &self.buf[..self.len()]
+        // SAFETY: `self.buf` is initialized to the correct length in `new()`.
```

Comment:
the purpose of this to avoid a bounds check? it's worth adding a comment about why the safe thing was not done.

Reply:
Yes, the purpose of using `unsafe` here is to avoid the bounds check for a minor performance improvement. However, it's a good point that adding a comment to explain this decision would be helpful. 

Now this that is not on the hot path for writing, I am less concerned and would be ok with going back to safe version as I didn't see a significant improvement. I do think its correct in this case, more of a matter of taste whether the unsafe belongs in this higher level api. I'll add a comment, but at your discretion I can revert back to the old version.